File: Algorithms/088.MergeSortedArray.py
class Solution:
    # @param A  a list of integers
    # @param m  an integer, length of A
    # @param B  a list of integers
    # @param n  an integer, length of B
    # @return nothing(void)
    def merge(self, A, m, B, n):
        i, j = 0, 0
        C = []
        while i < m and j < n:
            if A[i] < B[j]:
                C.append(A[i])
                i += 1
            else:
                C.append(B[j])
                j += 1
        if i < m:
            C.extend(A[i:])
        if j < n:
            C.extend(B[j:])
        for i in range(m+n):
            A[i] = C[i]
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: void Do not return anything, modify nums1 in-place instead.
        """
        while n >= 1:
            if m == 0 or nums1[m-1] <= nums2[n-1]:
                nums1[m+n-1] = nums2[n-1]
                n -= 1
            else:
                nums1[m+n-1] = nums1[m-1]
                m -= 1

File: Algorithms/cpp/088.MergeSortedArray.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int C[m+n];
        int i = 0;
        int j = 0;
        int k = 0;
        while (i < m && j < n) {
            if (nums1[i] < nums2[j]) {
                C[k++] = nums1[i++];
            }
            else {
                C[k++] = nums2[j++];
            }
        }
        while (i < m) {
            C[k++] = nums1[i++];
        }
        while (j < n) {
            C[k++] = nums2[j++];
        }
        k = 0;
        while (k < (m+n)) {
            nums1[k] = C[k];
            k++;
        }
    }
};
